crates/kornia-pnp/src/ops.rs

```diff
@@ -55,52 +55,131 @@ pub(crate) fn project_sq_error(
     Some(du.mul_add(du, dv * dv))
 }
 
+const EPSILON: f32 = 1e-10;
+const NUM_CONTROL_POINTS: usize = 4;
+const MAX_ITERATIONS: usize = 6;
+const PAIRS: [(usize, usize); 6] = [(0, 1), (0, 2), (0, 3), (1, 2), (1, 3), (2, 3)];
+const NUM_PAIRS: usize = PAIRS.len(); // 6
+
+/// Solves the linear system A * x = b for a 4x4 symmetric positive-definite matrix A
+/// using an unrolled Cholesky decomposition.
+///
+/// # Arguments
+/// * `a` - A reference to a 4x4 matrix, assumed to be symmetric positive-definite.
+/// * `b` - A reference to a 4x4 vector.
+///
+/// # Returns
+/// * `Some(Vector4<f32>)` containing the solution vector `x` if `A` is positive-definite.
+/// * `None` if the decomposition fails (i.e., `A` is not positive-definite).
+#[inline(always)]
+pub fn solve_4x4_cholesky(a: &Matrix4<f32>, b: &Vector4<f32>) -> Option<Vector4<f32>> {
```

Comment:
why not using this https://docs.rs/nalgebra/latest/nalgebra/linalg/struct.Cholesky.html ? or make our glam version ? I prefer the second